stacks/odincal_stack.py

```diff
@@ -0,0 +1,116 @@
+        pg_db_ssm_name: str,
+        psql_bucket_name: str,
+        queue_retention_period: Duration = Duration.days(14),
+        message_timeout: Duration = Duration.hours(24),
+        message_attempts: int = 4,
+        lambda_timeout: Duration = Duration.seconds(900),
+        **kwargs
+    ) -> None:
+        super().__init__(scope, id, **kwargs)
+
+        vpc = Vpc.from_lookup(
+            self,
+            "OdinSMROdincalVPC",
+            is_default=False,
+            vpc_name="OdinVPC",
+        )
+        vpc_subnets = SubnetSelection(
+            subnet_type=SubnetType.PRIVATE_WITH_EGRESS
+        )
+
+        queue_name = "OdinSMROdincalNotificationQueue"
+        notification_queue = Queue(
+            self,
+            queue_name,
+            queue_name=queue_name,
+            retention_period=queue_retention_period,
+            visibility_timeout=message_timeout,
+            content_based_deduplication=True,
+            dead_letter_queue=DeadLetterQueue(
+                max_receive_count=message_attempts,
+                queue=Queue(
+                    self,
+                    "Failed" + queue_name,
+                    queue_name="Failed" + queue_name,
+                    retention_period=queue_retention_period,
+                    content_based_deduplication=True,
+                ),
+            ),
+        )
+
+        level1_lambda = DockerImageFunction(
+            self,
+            "OdinSMROdincalLambda",
+            code=DockerImageCode.from_image_asset(
+                "./odincal",
+            ),
+            vpc=vpc,
+            vpc_subnets=vpc_subnets,
+            timeout=lambda_timeout,
+            architecture=Architecture.X86_64,
+            ephemeral_storage_size=Size.mebibytes(512),
+            memory_size=1024,
+            environment={
+                "ODIN_PG_HOST_SSM_NAME": pg_host_ssm_name,
+                "ODIN_PG_USER_SSM_NAME": pg_user_ssm_name,
+                "ODIN_PG_PASS_SSM_NAME": pg_pass_ssm_name,
+                "ODIN_PG_DB_SSM_NAME": pg_db_ssm_name,
+                "ODIN_PSQL_BUCKET_NAME": psql_bucket_name,
+                "ODIN_L1_NOTIFICATIONS": notification_queue.queue_name,
+            },
+        )
+
+        for ssm_name in (
+            pg_host_ssm_name,
+            pg_user_ssm_name,
+            pg_pass_ssm_name,
+            pg_db_ssm_name,
+        ):
+            level1_lambda.add_to_role_policy(PolicyStatement(
+                effect=Effect.ALLOW,
+                actions=["ssm:GetParameter"],
+                resources=[f"arn:aws:ssm:*:*:parameter{ssm_name}"]
+            ))
+
+        psql_bucket = Bucket.from_bucket_name(
+            self,
+            "OdinSMROdincalPsqlBucket",
+            psql_bucket_name,
+        )
+
+        psql_bucket.grant_read(level1_lambda)
+
+        l0_jobs_queue = Queue.from_queue_arn(
+            self,
+            "OdinSMRLevel0JobsQueue",
+            Fn.import_value(level0_queue_arn),
```

Comment:
is level0_queue_arn really the arn? In app.py it looks like a name?

Reply:
It is supposed to be the ARN, but I don't know what the ARN will be yet, since that queue will be created in L0, so the value in `app.py` is a placeholder. This method is copied from the L0-fetcher and RAC-lambda interaction in MATS.

Reply:
It would be nicer if you could get the queue from the name, but as far as I've been able to tell that is not supported.